### PR TITLE
s3zip emitter - zip records before emitting to s3

### PR DESCRIPTION
**Pull request by workanandr**
Implementation of S3ZipEmitter: Compress the records into a zip file before emitting.

### CODE REVIEW

File: src/main/java/com/amazonaws/services/kinesis/connectors/s3/S3ZipEmitter.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                                + Arrays.toString(record),
+                        e);
+                return buffer.getRecords();
+            }
+        }
+
+        // Get the Amazon S3 filename
+        String s3FileName = super.getS3FileName(buffer.getFirstSequenceNumber(), buffer.getLastSequenceNumber());
+
+        compressedRecords.add(this.compress(s3FileName, baos.toByteArray()));
```
Comment (darkcrawler01) [inline src/main/java/com/amazonaws/services/kinesis/connectors/s3/S3ZipEmitter.java]: are you able to read the file ?

Comment (workanandr) [inline src/main/java/com/amazonaws/services/kinesis/connectors/s3/S3ZipEmitter.java]: yes.

Comment (darkcrawler01) [inline src/main/java/com/amazonaws/services/kinesis/connectors/s3/S3ZipEmitter.java]: can you share code for the file reader ?
does redshift COPY command work with this file ?

Comment (workanandr) [inline src/main/java/com/amazonaws/services/kinesis/connectors/s3/S3ZipEmitter.java]: Sorry, I misunderstood your earlier question. The redshiftbasic emitter will not work with this. It needs to be modified to uncompress before reading the file. 
My main intention was not specifically for use with redshift connector. Rather a basic s3 connector where raw data can be archived. If manifest functionlity is added (straight forward), other lazy consumers can read from s3 and do further processing.

**Pull request closed without merging**